gcp_variant_transforms/pipeline_common.py

```diff
@@ -69,33 +69,62 @@ def parse_args(argv, command_line_options):
   return known_args, pipeline_args
 
 
-def get_pipeline_mode(input_pattern, optimize_for_large_inputs=False):
+def get_pipeline_mode(
+    input_patterns,
+    input_file=None,
+    optimize_for_large_inputs=False):
   # type: (str, bool) -> int
   """Returns the mode the pipeline should operate in based on input size."""
   if optimize_for_large_inputs:
     return PipelineModes.LARGE
 
-  match_results = filesystems.FileSystems.match([input_pattern])
+  match_results = filesystems.FileSystems.match(input_patterns)
   if not match_results:
-    raise ValueError('No files matched input_pattern: {}'.format(input_pattern))
+    if input_file:
+      raise ValueError(
+          'No files matched any input pattern in {} file'.format(input_file))
+    else:
+      raise ValueError(
+          'No files matched {} input pattern'.format(input_patterns[0]))
+
+  total_files = 0
+  for match in match_results:
+    if not match.metadata_list:
+      raise ValueError(
+          'Input pattern {} did not match any files.'.format(match.pattern))
+    total_files += len(match.metadata_list)
 
-  total_files = len(match_results[0].metadata_list)
   if total_files > _LARGE_DATA_THRESHOLD:
     return PipelineModes.LARGE
   elif total_files > _SMALL_DATA_THRESHOLD:
     return PipelineModes.MEDIUM
   return PipelineModes.SMALL
 
 
-def read_headers(pipeline, pipeline_mode, input_pattern):
+def get_file_names(input_file):
+  # type (str) -> List(str)
+  """ Reads all input file and extracts list of patterns out of it."""
```

Comment:
All this does it takes a file and returns a list of the lines, correct?  Is there no python builtin for this?  (something like  open("foo").readlines() or so?)